workflow-trace-viewer/src/jvmMain/kotlin/com/squareup/workflow1/traceviewer/App.kt

```diff
@@ -1,9 +1,44 @@
 package com.squareup.workflow1.traceviewer
 
+import androidx.compose.foundation.layout.Box
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import io.github.vinceglb.filekit.PlatformFile
+import io.github.vinceglb.filekit.readString
 
+/**
+ * Main composable that provides the different layers of UI.
+ */
 @Composable
-fun App() {
-  Text("Hello world!")
+public fun App(
+  modifier: Modifier = Modifier
+) {
+  Box {
+    val selectedFile = remember { mutableStateOf<PlatformFile?>(null) }
+
+    if (selectedFile.value != null) {
+      SandboxBackground { WorkflowContent(selectedFile.value) }
+    }
+
+    UploadFile(onFileSelect = { selectedFile.value = it })
+  }
+}
+
+@Composable
+private fun WorkflowContent(file: PlatformFile?) {
```

Comment:
Nit: No need for `file` to be nullable here. I imagine it's because Kotlin thinks `.value` can change but we know it won't so it's okay to use `!!` here to force it to be non-null.